### ISSUE TITLE
[Feature Request]: Programmatically control selectedItems in Filterable MultiSelect

### ISSUE THREAD

**Issue by ask-42** [carbon-design-system/carbon#10340]
### Summary

The non-filterable MultiSelect was enhanced to allow you to pass `selectedItems` to fully control it (https://github.com/carbon-design-system/carbon/pull/10236).

It would be nice if the Filterable MultiSelect allowed this as well, as we need this functionality. I currently work around this issue with a hacky solution involving subclassing the component, but I'd prefer not to do that, and it won't be an option anymore once it has been migrated to being a functional component. Right now it only supports `initialSelectedItems`, but I need full control over the selection even after mounting.

### Required functionality

`selectedItems` can be programmatically controlled.

### Specific timeline issues / requests

Before or alongside any release that migrates the component to be functional instead of class-based.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

**Comment by abbeyhrt** [comment]
We would love to have this contributed! We think it's a great feature to have and similar to MultiSelect, we would love to help out anyone who would want to contribute this solution!

**Comment by sstrubberg** [comment]
Per @tw15egan's comments during Backlog Cleaning this morning. We would need look at mutliselect and see if we can port the changes made in #4812 to filterable multi-select. Possible design guidance based on this. It could be a healthy amount of sprint work, if mutli-select is portable. But if not, it could take much longer.

**Comment by marconuccetelli** [comment]
@mthaak's solution works for updating data, but it causes the list to scroll up whenever something changes within the component. Any news on this functionality?

**Comment by sstrubberg** [comment]
Unfortunately, no movement has been made by the maintainer team on this request. Due to roadmap constraints, we've identified this as work we would love to see the community take up by creating a PR. We'll let you know if anything changes!

**Comment by ychavoya** [comment]
Hi, I'd like to be assigned this issue is possible to work on it, thanks!

**Comment by tw15egan** [comment]
@ychavoya would https://github.com/carbon-design-system/carbon/pull/16139 also resolve this issue?

**Comment by ychavoya** [comment]
@tw15egan No, the Filterable MultiSelect does not have any implementation on controlled input. I actually ran into the issue that that PR solves by trying to explore this feature

**Comment by tw15egan** [comment]
Ah, my mistake, @ychavoya; I did not see the distinction that this issue is regarding `FilterableMultiselect.`

**Comment by ychavoya** [comment]
Hi, just FYI I noticed that the changes from #16147 may add this functionality since it seems to implement the `useSelection` hook used in the regular Multiselect. It may be worth to check when that PR is done
